azure-pipelines.yml

```diff
@@ -99,6 +99,21 @@ jobs:
     parameters:
       jobTemplate: build-job.yml
       buildConfig: debug
+      ${{ if eq(variables['Build.DefinitionName'], 'coreclr-ci') }}:
+        platforms:
+        - Windows_NT_x64
+        - Windows_NT_x86
+      ${{ if eq(variables['Build.DefinitionName'], 'coreclr-outerloop') }}:
```

Comment:
This wouldn't work - see top level `${{ if }}` which restricts to coreclr-ci build definition

You have two options:
1) relax top level if-predicate (i.e. include coreclr-outerloop in the list of allowed build definitions)
2) clone the whole code block and have two separate blocks for debug build-job - one for coreclr-ci and another for coreclr-outerloop